src/main/java/io/vertx/core/Expectation.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2011-2019 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.core;
+
+@FunctionalInterface
+public interface Expectation<T> extends Assertion<T> {
+
+  boolean test(T t);
+
+  @Override
+  default void check(T t) throws Throwable {
+    if (!test(t)) {
+      throw new AssertionError(toException(t));
```

Comment:
When an `Expectation` is created using a lambda expression, which is likely to be very common, `toException` will be `null` and hence we'll `throw new AssertionError(null)`, which is not very friendly.

Wouldn't something like this be better?

```java
default void check(T t) throws Throwable {
  if (!test(t)) {
    Exception exception = toException(t);
    if (exception == null) {
      throw new AssertionError("Expectation failed for " + t);
    } else {
      throw new AssertionError(exception); // or just `throw exception`?
    }
  }
}
```

Reply:
OK, I see that the `Future` operation that handles an `Expectation` deals with this situation, so please just ignore me here :-)